test/python/test_localjob.py

```diff
@@ -15,188 +15,154 @@
-            if all([job.done for job in job_array]):
-                self.log.warning('all jobs completed before simultaneous jobs '
-                                 'could be detected')
-                break
-            for job in job_array:
-                self.log.info('%s %s %s', job.status['status'], job.cancelled,
-                              check)
-            self.log.info('{0} {1:0.2f}'.format('-'*20, time.time()-start_time))
-            if time.time() - start_time > timeout:
-                raise TimeoutError('failed to see multiple running jobs after '
-                                   '{0} s'.format(timeout))
-            time.sleep(1)
-
-        # Wait for all the jobs to finish.
-        _ = [job.result() for job in job_array if not job.cancelled]
+
+        self.assertCalledOnce(executor.submit)
+        mockedFuture = executor.submit.return_value
+        self.assertCalledOnce(mockedFuture.cancel)
 
     def test_done(self):
-        backend = self._provider.get_backend('local_qasm_simulator_py')
-        qobj = qiskit._compiler.compile(self._qc, backend)
-        quantum_job = QuantumJob(qobj, backend, preformatted=True)
-        job = backend.run(quantum_job)
-        job.result()
-        self.assertTrue(job.done)
-
-    def test_get_backend_name(self):
-        backend_name = 'local_qasm_simulator_py'
-        backend = self._provider.get_backend(backend_name)
-        qobj = qiskit._compiler.compile(self._qc, backend)
-        quantum_job = QuantumJob(qobj, backend, preformatted=True)
-        job = backend.run(quantum_job)
-        self.assertTrue(job.backend_name == backend_name)
+        # Like before.
+        # pylint: disable=redefined-outer-name
+        with intercepted_executor_for_LocalJob() as (LocalJob, executor):
+            job = LocalJob(lambda: None, FakeQJob())
+            _ = job.done
+
+        self.assertCalledOnce(executor.submit)
+        mockedFuture = executor.submit.return_value
+        self.assertCalledOnce(mockedFuture.done)
+
+    def assertCalledOnce(self, mockedCallable):
+        """Assert a mocked callable has been called once."""
+        call_count = mockedCallable.call_count
+        self.assertEqual(
+            call_count, 1,
+            'Callable object has been called more than once ({})'.format(
+                call_count))
+
+
+class FakeQJob():
+    def __init__(self):
+        self.backend = FakeBackend()
+        self.qobj = {
+            'id': 'test-id',
+            'config': {
+                'backend_name': self.backend.name,
+                'shots': 1024,
+                'max_credits': 100
+            },
+            'circuits': [{
+                'compiled_circuit_qasm': 'fake-code',
+                'config': {
+                    'seed': 123456
+                }
+            }]
+        }
+
+
+class FakeBackend():
+
+    def __init__(self):
+        self.name = 'test-backend'
+
+
+@contextmanager
+def intercepted_executor_for_LocalJob():
```

Comment:
Can you rename this method to snake case (`intercepted_executor_for_localjob`)? There are a number of other variable names that have also been hidden by the `disable=invalid-name` directive at the top (`backendConstructor`, `mockedFuture`, `mockedCallable`)  - in the tests we are not too strict about the linter (as, for example, the `assertCalledOnce` is a valid name despite nor being technically in line with the linter preferences), but in general we should adhere to those.

Reply:
The case for `assertCalledOnce` is that it adheres to the API convention for assertion ids in the `Test` class but I can change it if needed.

Reply:
Sure - I  was actually trying to say that `assertCalledOnce` is an example of a reasonable exception to the linter rule and makes sense to keep it in camel-case :)